tests/variantstudy/model/test_variant_model.py

```diff
@@ -190,3 +199,126 @@ def test_commands_service(
+        command_7 = CommandDTO(
+            action="update_comments", args={"comments": "another new comment"}, study_version=study_version
+        )
+
+        variant_study_service.append_command(variant_id, command_6, params=owner_params)
+        variant_study_service.append_command(variant_id, command_7, params=user2_params)
+
+        # Make sure there are commands generated by both users
+        commands = variant_study_service.get_commands(variant_id, params=owner_params)
+        assert len(commands) == 2
+
+        # Make sure their `user_name` and `updated_at` attributes are not None
+        for command in commands:
+            assert command.user_name and command.updated_at
+
+        # Make sure commands has not the same author
+        assert commands[0] != commands[1]
+        assert commands[0].user_name == "john.doe"
+        assert commands[1].user_name == "jane.doe"
+
+    @pytest.mark.parametrize("root_study_id", [False], indirect=True)
+    @with_db_context
+    def test_command_same_author(
+        self,
+        jwt_user: JWTUser,
+        variant_study_service: VariantStudyService,
+        root_study_id: str,
+    ):
+        """
+        Test the case of multiple commands was created by the same user.
+        Set up:
+            Initialize a counter of queries to database
+            Define a watcher on the orm queries to database that updates the counter
+            Create a user
+            Create a variant study
+            Make the user generates five commands on the newly created variant
+        Test:
+            Each time a command is retrieved, the database must be accessed only if
+            the author of the currently retrieved command is not already known during
+            the process
+        """
+        nb_queries = 0  # Store number of orm queries to database
+
+        # Watch orm events and update `nb_queries`
+        @event.listens_for(db.session, "do_orm_execute")
+        def check_orm_operations(orm_execute_state):
+            if orm_execute_state.is_select:
+                nonlocal nb_queries
+                nb_queries += 1
+
+        owner_params = RequestParameters(user=jwt_user)
+
+        # Generate a variant on a study that allow other user to edit it
+        variant_study = variant_study_service.create_variant_study(root_study_id, "new_variant", params=owner_params)
+
+        commands = []
+
+        # Create two new commands on the existing variant
+        for index in range(5):
+            commands.append(
+                CommandDTO(
+                    action="update_comments",
+                    args={"comments": f"new comment {index}"},
+                    study_version=StudyVersion.parse(variant_study.version),
+                )
+            )
+
+        variant_study_service.append_commands(variant_study.id, commands, params=owner_params)
+
+        nb_queries_before = nb_queries  # store initial state
+
+        with patch.object(
```

Comment:
Now that you introduced the orm listener I think you can remove this as it's an implementation detail